src/hidra/receiver/plugins/asapo_producer.py

```diff
@@ -120,6 +118,17 @@ def check_config(logger, config, required_parameter):
         raise utils.WrongConfiguration("The configuration has missing or wrong parameters.")
 
 
+def parse_file_path(file_regex, file_path):
+    # parse file name
+    search = re.search(file_regex, file_path)
```

Comment:
This compiles the regex for each file. Would be more efficient to compile it once and reuse the compiled version

Reply:
How, if each time file_path is different?

Reply:
```
# do this only if file_regex changes
compiled_file_regex = re.compile(file_regex)
# do this for every file_path
compiled_file_regex.search(file_path)
```

Reply:
Both works very quick. It increases complexity but not clear, if we can gain something out of it.

Reply:
The complexity increase is minimal. Just do `self.file_regex = re.compile(file_regex)` in `__init__` change
```
search = file_regex.search(file_path)
...
logger.debug("file_regex: %s", file_regex.pattern)